### ISSUE TITLE
Pylint 2.5 does not allow ignoring checkers whose names have numbers in them

### ISSUE THREAD

**Issue by jfly** [PyCQA/pylint#3666]
### Steps to reproduce
Using a version of Pylint 2.5, it's impossible to ignore warnings about checkers whose names contain numbers.

Reproducing this requires writing custom checker, which is a bit complicated to describe in a single issue. I've put together a simple reproduction here: https://github.com/jfly/pylint-demo#pylint-checker-naming-issue

### Current behavior

In the demo above, a line like this should pass pylint, but instead I get a `Bad option value '-no-jfly' (bad-option-value)` message. Note that the option value is actually `j3-no-jfly`, *not* `-no-jfly`:

```python
pylint: disable=j3-no-jfly
```

### Expected behavior

I suppose it's possible that Pylint intentionally does not want to support checkers with numbers in their names. That used to be supported, though, so I'd prefer that Pylint support this (as it did in 2.4 and before).

If it *is* intentional to not support numbers in checker names, then I think it would be nice for this to be explicitly called out in a release note, and I also think the error could be clearer about what's going on.

### pylint --version output

(see [readme](https://github.com/jfly/pylint-demo#to-reproduce-the-issue-using-pylint-250) for more information)

```
$ pylint --version
pylint 2.5.0
astroid 2.4.1
Python 3.8.3 (default, May 17 2020, 18:15:42)
[GCC 10.1.0]
```

### Some digging into how this broke between pylint 2.4.4 and 2.5

As demonstrated [here](https://github.com/jfly/pylint-demo), this worked on Pylint 2.4.4, and started breaking with Pylint 2.5.0. I did some code spelunking, and it looks like this commit https://github.com/PyCQA/pylint/commit/9bdae8b82fcc5b0592135cbf6bead7df360a6672 broke support for checkers with numbers in the name. Specifically, it defines `MESSAGE_STRING`'s regex in a way such that numbers are not allowed. One fix may be to just tweak that regex so it allows numbers.
